pay-api/src/pay_api/services/non_sufficient_funds.py

```diff
@@ -0,0 +1,192 @@
+        return self.__dao
+
+    @_dao.setter
+    def _dao(self, value):
+        self.__dao = value
+        self.id: int = self._dao.id
+        self.invoice_id: int = self._dao.invoice_id
+        self.description: str = self._dao.description
+
+    @property
+    def id(self):
+        """Return the _id."""
+        return self._id
+
+    @id.setter
+    def id(self, value: int):
+        """Set the _id."""
+        self._id = value
+        self._dao.id = value
+
+    @property
+    def invoice_id(self):
+        """Return the _invoice_id."""
+        return self._invoice_id
+
+    @invoice_id.setter
+    def invoice_id(self, value: int):
+        """Set the _invoice_id."""
+        self._invoice_id = value
+        self._dao.invoice_id = value
+
+    @property
+    def description(self):
+        """Return the Non-Sufficient Funds description."""
+        return self._description
+
+    @description.setter
+    def description(self, value: str):
+        """Set the Non-Sufficient Funds description."""
+        self._description = value
+        self._dao.description = value
+
+    def commit(self):
+        """Save the information to the DB."""
+        return self._dao.commit()
+
+    def rollback(self):
+        """Rollback."""
+        return self._dao.rollback()
+
+    def flush(self):
+        """Save the information to the DB."""
+        return self._dao.flush()
+
+    def save(self):
+        """Save the information to the DB."""
+        return self._dao.save()
+
+    def asdict(self):
+        """Return the Non-Sufficient Funds as a python dict."""
+        non_sufficient_funds_schema = NonSufficientFundsSchema()
+        d = non_sufficient_funds_schema.dump(self._dao)
+        return d
+
+    @staticmethod
+    def populate(value: NonSufficientFundsModel):
+        """Populate Non-Sufficient Funds Service."""
+        non_sufficient_funds_service: NonSufficientFundsService = NonSufficientFundsService()
+        non_sufficient_funds_service._dao = value  # pylint: disable=protected-access
+        return non_sufficient_funds_service
+
+    @staticmethod
+    def query_all_non_sufficient_funds_invoices(account_id: str, page: int, limit: int):
+        """Return all Non-Sufficient Funds invoices."""
+        query = db.session.query(PaymentModel, InvoiceModel) \
+            .join(PaymentAccountModel, PaymentAccountModel.id == PaymentModel.payment_account_id) \
+            .outerjoin(InvoiceReferenceModel, InvoiceReferenceModel.invoice_number == PaymentModel.invoice_number) \
+            .outerjoin(InvoiceModel, InvoiceReferenceModel.invoice_id == InvoiceModel.id) \
+            .join(NonSufficientFundsModel, InvoiceModel.id == NonSufficientFundsModel.invoice_id) \
+            .filter(PaymentAccountModel.auth_account_id == account_id) \
+            .filter(PaymentModel.paid_amount > 0)
+
+        query = query.order_by(PaymentModel.id.asc())
+        pagination = query.paginate(per_page=limit, page=page)
```

Comment:
This paginate looks at part of the request, just a heads up:

![image](https://github.com/bcgov/sbc-pay/assets/3484109/94d3bbb7-84ec-42f1-9ba5-95b325c49ce3)

Reply:
Probably need something like this:

```
if page or limit:
 results, total = query.paginate(per_page=limit, page=page)
else 
  invoices_nsf = items.all()
  results, total = invoice_nsf, len(invoices_nsf)
  ```